js/index.js

```diff
@@ -37,26 +32,31 @@ exports = new (Class(function() {
     if (!v.failed) {
       received_data = JSON.parse(v.notification_data);
       logger.log("{onesignal} data at js", JSON.stringify(v));
-      invokeCallbacks(cb, received_data, "NotificationReceived");
+      invokeCallback("NotificationReceived", received_data);
     }
   });
 
   NATIVE.events.registerHandler('onesignalNotificationOpened', function(v) {
     var received_data;
 
     received_data = JSON.parse(v.notification_data);
-    logger.log("{onesignal} data at js", JSON.stringify(v));
-    invokeCallbacks(cb, received_data, "NotificationOpened");
+    logger.log("{onesignal} onOpened data at js", JSON.stringify(v));
+    invokeCallback("NotificationOpened", received_data);
   });
 
   // SendTags
   this.sendTags = function (obj) {
     pluginSend('sendUserTags', obj);
   };
 
-  this.registerCallback = function (next) {
-    if(cb.length < 1) {
-      cb.push(next);
+  this.registerCallback = function (callback) {
+
+    cb = callback;
+    for (var key in data) {
```

Comment:
data is declared on top and value is assigned in invokeCallbacks.

If a callback is registered late, then the data is stored in a variable and then the callback is invoked with the data from push notifications